src/test/scala/io/iohk/ethereum/jsonrpc/server/http/JsonRpcHttpServerSpec.scala

```diff
@@ -242,6 +244,116 @@ class JsonRpcHttpServerSpec extends AnyFlatSpec with Matchers with ScalatestRout
     }
   }
 
+  it should "return status code OK when throw LogicError" in new TestSetup {
+    val jsonRpcError = JsonRpcError.LogicError("Faucet error: Connection not established")
+    (mockJsonRpcController.handleRequest _)
+      .expects(*)
+      .returning(
+        Task.now(
+          JsonRpcResponse(
+            "2.0",
+            None,
+            Some(JsonRpcError(code = jsonRpcError.code, message = jsonRpcError.message, data = None)),
+            JInt(1)
+          )
+        )
+      )
+
+    val jsonRequest = ByteString("""{"jsonrpc":"2.0", "method": "asd", "id": "1"}""")
+    val postRequest =
+      HttpRequest(HttpMethods.POST, uri = "/", entity = HttpEntity(MediaTypes.`application/json`, jsonRequest))
+
+    postRequest ~> Route.seal(mockJsonRpcHttpServer.route) ~> check {
+      status shouldEqual StatusCodes.OK
+      responseAs[String] shouldEqual
```

Comment:
What do you think if instead of comparing Strings (which in case of an error forces the developer to compare the whole string to find the difference), we parse the response as Json and assert each attribute individually?

Reply:
Done